source/src/console.cpp

```diff
@@ -585,6 +585,8 @@ void consolekey(int code, bool isdown, int cooked, SDLMod mod)
                 h->run();
                 if(h->action && !storeinputcommand) history.drop();
             }
+
+            delete h;
```

Comment:
There is a memory leak, but this line will lead to a double free.
the delete h should be only if we do history.drop()

Reply:
Its been almost two years since I've open this pull request and at the time I was very eager to get anything accepted into a project that I enjoyed like AssaultCube but now reviewing my own changes after all this time I can safely say they do not add anything of worth and substance to be merged in with this project. I played this game for the better part of two or three years before I moved onto CSGO and devoted my attention to that instead. AssaultCube will always have a soft spot in my heart when it comes to FPS games as it was my first online computer game and FPS. Maybe one day I can finally give something back to the community but I don't know for sure as the code for AssaultCube is, to put it mildly, a little messy. :P

You can close this pull request. Bye!

Reply:
I totally agree.
though this is a good project to give to students to find memory leaks, refactor code
and most importantly see what a messy code can do to you...
Which is exactly what I did.
Would be nice to get this project active again, nay be convert it to C++17 and refactor the
hell out it.

Reply:
Definitely would be a great exercise and activity for a student contingent to learn how to do all of that listed above for sure. But I don't know if it is the norm for a teacher to find inactive projects and start having the students work on it. Maybe students at an individual level could participate? Not sure at all as I don't know who would want to work on such a degraded code base. But it sure is a good example of what badly written code can do to someone. Was the code this bad always? Briefly, I checked another successor of the Cube FPS, Cube 2: Sauerbraten and I didn't check everything but the code looked much more clean. At least that was my initial impression from my glance at it.